### PR TITLE
feature-add-ui-profile-photo-in-profilesettings-#24

### PR DESCRIPTION
**Pull request by Zorba11**
### What this PR does (required):
UI for uploading user's profile picture.

### Screenshots / Videos (front-end only):
Screen Shot 2021-10-16 at 7 02 06 PM
Screen Shot 2021-10-16 at 7 31 12 PM

### Any information needed to test this feature (required):

1. run the react app locally
2. login with credentials 
3. goto `localhost:3000/settings` you can see the UI

### Any issues with the current functionality (optional):

1. There is a bug fix related to routing. Please check that as well, I might be wrong (Login.tsx & useAuthContext.tsx)
2. As I mentioned during the standup I've accidentally started building the NavBart and the SideDraw components (1st screenshot). I'll be using the same code in a future ticket. for now I'm committing & deleting those components for future reference. So, only what you see in the 2nd screenshot will be pushed with this PR

### CODE REVIEW

File: client/src/components/PhotoUpload/PhotoUpload.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import useStyles from './useStyles';
+import { Card, Typography, Avatar, Button } from '@material-ui/core';
+import DeleteIcon from '@material-ui/icons/Delete';
+import avatarImage from '../../Images/b1f0e680702e811aa8ba333cb19c0e0ea95e8e31.png';
```
Comment (Zorba11) [inline client/src/components/PhotoUpload/PhotoUpload.tsx:5]: This is just a place holder image until `Setup/IN: Amazon S3 or Cloudinary #41` is done.

File: client/src/components/PhotoUpload/useStyles.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        marginTop: '20%',
+        top: '32%',
+        width: '700px',
+      },
+      height: '570px',
+      margin: 8,
+      transform: 'translate(-50%,-50%)',
+      position: 'absolute',
+      marginTop: '26%',
+      // top: '80%',
```
Comment (ezeilo-su) [inline client/src/components/PhotoUpload/useStyles.ts:28]: Remove unused code

File: client/src/pages/ProfileSettings/ProfileSettings.test.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      <MockAuthProvider>
+        <ProfileSettings />
+      </MockAuthProvider>,
+    );
+    expect(screen.getByText('Profile Photo')).toBeInTheDocument();
+    expect(screen.getByText('Delete photo')).toBeInTheDocument();
+    expect(screen.getByText('Upload a file from your device')).toBeInTheDocument();
+    expect(screen.getByRole('upload-avatar')).toBeInTheDocument();
+  });
+});
```
Comment (ezeilo-su) [inline client/src/pages/ProfileSettings/ProfileSettings.test.tsx:26]: :+1:

**Review by ezeilo-su — COMMENTED** [review]
This one looks good! Just a minor comment below.
Happy coding!

File: client/src/components/PhotoUpload/useStyles.ts (hunk 2 above)
Comment (Zorba11) [reply client/src/components/PhotoUpload/useStyles.ts:28]: @ezeilo-su sorry about that. will do ! thank you !

Comment (Zorba11) [reply client/src/components/PhotoUpload/useStyles.ts]: Hi, just made the change

File: client/src/components/PhotoUpload/PhotoUpload.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+const PhotoUpload = (): JSX.Element => {
+  const classes = useStyles();
+  const infoString = 'Be sure to use a photo that clearly shows your face';
```
Comment (rajivtitus) [inline client/src/components/PhotoUpload/PhotoUpload.tsx:9]: 👍🏽

File: client/src/pages/ProfileSettings/ProfileSettings.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  const history = useHistory();
+
+  if (loggedInUser === undefined) return <CircularProgress />;
+  if (!loggedInUser) {
+    history.push('/login');
+    return <CircularProgress />;
+  }
+
+  return (
+    <div className={classes.root}>
```
Comment (rajivtitus) [inline client/src/pages/ProfileSettings/ProfileSettings.tsx:21]: You can consider using the MUI 'Box' component here :)

Ref: https://mui.com/components/box/

Eg: `<Box display="flex" alginItems="center"....`

File: client/src/components/PhotoUpload/PhotoUpload.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  return (
+    <Card className={classes.root}>
+      <Typography variant="body2" className={classes.title}>
```
Comment (rajivtitus) [inline client/src/components/PhotoUpload/PhotoUpload.tsx:12]: Optional: For increase semantics you can include the content within the 'CardContent' component and the actions within the 'CardActions' component :)

**Review by rajivtitus — CHANGES_REQUESTED** [review]
Great work! The UI on this looks good, just a couple of minor comments to address.
Some comments have been labelled as 'optional', you can make a decision to implement these changes if it's not too time intensive :)

**Review by rajivtitus — APPROVED** [review]

File: client/src/pages/ProfileSettings/ProfileSettings.tsx (hunk 5 above)
Comment (Zorba11) [reply client/src/pages/ProfileSettings/ProfileSettings.tsx:21]: 👍

**Pull request merged**